drools-commands/src/main/java/org/drools/commands/runtime/BatchExecutionCommandImpl.java

```diff
@@ -187,4 +187,14 @@ public String toString() {
                ", commands=" + commands +
                '}';
     }
+
+    @Override
+    public boolean autoFireAllRules() {
+        for ( Command nestedCmd : getCommands() ) {
+            if ( nestedCmd instanceof FireAllRulesCommand ) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
Do you really need to consider `FireAllRulesCommand` as a special case? Isn't this code equivalent
```suggestion
        return getCommands().stream().allMatch(Command::autoFireAllRules);
```

Note: I wrote it by hearth so it might not compile :)